src/components/app/list-new/AppList.tsx

```diff
@@ -81,491 +97,248 @@ export default function AppList({ isSuperAdmin, appListCount, isArgoInstalled }:
-        // return if not single cluster
-        if (clusterFilters.length != 1) {
-            return
+    const getFormattedFilterValue = (filterKey: AppListUrlFilters, filterValue: string) => {
+        switch (filterKey) {
+            case AppListUrlFilters.cluster:
+                return appListFilterResponse?.result?.Clusters.find((cluster) => cluster.id === +filterValue)
+                    ?.cluster_name
+            case AppListUrlFilters.project:
+                return +filterValue
+                    ? appListFilterResponse?.result?.Teams.find((team) => team.id === +filterValue)?.name
+                    : 'Apps with no project'
+            case AppListUrlFilters.environment:
+                return appListFilterResponse?.result?.Environments.find((env) => env.id === +filterValue)
+                    ?.environment_name
+            case AppListUrlFilters.namespace:
+                return filterValue.split('_')[1]
+            case AppListUrlFilters.templateType:
+                return filterValue === FluxCDTemplateType.HELM_RELEASE ? FLUX_CD_HELM_RELEASE_LABEL : filterValue
+            default:
+                return filterValue
         }
-
-        const _cluster = clusterFilters[0]
-
-        // return if any cluster filter applied
-        const _isAnyClusterFilterApplied = clusterFilters.some((_cluster) => _cluster.isChecked)
-        if (_isAnyClusterFilterApplied) {
-            return
-        }
-
-        // auto check cluster
-        const _filterOptions: FilterOption[] = []
-        _filterOptions.push({
-            key: _cluster.key,
-            label: _cluster.label,
-            isSaved: _cluster.isSaved,
-            isChecked: true,
-        })
-
-        applyFilter(AppListConstants.FilterType.CLUTSER, _filterOptions, currentTab)
     }
 
-    const onRequestUrlChange = (showExportCsvButton?: boolean): any => {
-        const searchQuery = location.search
-
-        const { payload, filterApplied } = getPayloadFromUrl(searchQuery, appCount, showExportCsvButton)
-
-        const _masterFilters = structuredClone(InitialEmptyMasterFilters)
-
-        // set projects (check/uncheck)
-        _masterFilters.projects = masterFilters.projects.map((project) => {
-            return {
-                key: project.key,
-                label: project.label,
-                isSaved: true,
-                isChecked: filterApplied.teams.has(project.key),
+    const projectOptions: GroupedOptionsType[] = [
+        { label: '', options: [APPS_WITH_NO_PROJECT_OPTION] },
+        appListFilterResponse?.result.Teams.length && {
+            label: 'Projects',
+            options: [
+                ...(appListFilterResponse?.result.Teams.map((team) => ({
+                    label: team.name,
+                    value: String(team.id),
+                })) ?? []),
+                ...(projectListResponse?.result.map((team) => ({ label: team.name, value: String(team.id) })) ?? []),
```

Comment:
Check if list is unique?

Reply:
only one of the list will be available at once